lib/templates/android_screen_base.tt

```diff
@@ -183,4 +183,55 @@ class AndroidScreenBase < Calabash::ABase
     # Clicking in the Done button
     touch "* id:'button1'"
   end
+  
+  # @param query [String] query that will be searched
+  # @param scroll_location_id [String]
+  # @param scroll_location_query [String, nil]
+  # @param direction [Symbol]
+  # @param options [Hash]
+  # @option options [String] :timeout
+  # @option options [String] :flick in case we need flick
+  # @todo #TODO Add reset option to allow going to top position
+  # before start search
+  # Exemple -> scroll_looking_for_query(query_text_variable, nil, scroll_location_query_variable, :down, :timeout=>7)
+
+  def scroll_looking_for_query query, scroll_location_id, scroll_location_query = nil , direction= :down, options = {}
+
+    if scroll_location_query.nil?
+      scroll_location_query="* id:'#{scroll_location_id}'"
+    end
+
+    if options[:timeout].nil?
+      options[:timeout]=15
+    end
+
+    if options[:flick].nil?
+      options[:flick]=false
+    end
+    wait_poll(:until_exists => query, :timeout => options[:timeout]) do
+      if options[:flick]
+        flick(scroll_location_query, direction)
+      else
+        scroll(scroll_location_query, direction)
+      end
+    end
+  end
+  
+  # @param [Hash] options it is possible to pass the view id or the query
+  # @option options [String] :id The view's id
+  # @option options [String] :query The complete view query
+  # @return [String] returns a string associated with a single view, if the case
+  # @return [Array] return an array[String] associated with the views, if the case
+  # Example ->     get_text_from(query: query_text_variable)
+  def get_text_from(options = {})
+    query_text = "* id:'#{options[:id]}'" unless options[:id].nil?
+    query_text = options[:query] unless options[:query].nil?
+    array_text = query(query_text, :text)
+
+    # returns an array[String] or text dependending of the quantity of views found 
+    return array_text.first unless array_text.size > 1
```

Comment:
Calabash answer will always be an array. If you do this, you will get two different kind of answers.
In this case will be the text, in the case below will be an array with the text in the first position.

Wouldn't it be better to use only `return array_text.first`?